provider/kubernetes.go

```diff
@@ -170,10 +170,16 @@ func (provider *Kubernetes) loadIngresses(k8sClient k8s.Client) (*types.Configur
 					log.Errorf("Error retrieving services %s", pa.Backend.ServiceName)
 				}
 				for _, service := range services {
-					var protocol string
+					protocol := "http"
 					for _, port := range service.Spec.Ports {
 						if port.Port == pa.Backend.ServicePort.IntValue() {
-							protocol = port.Name
+							if len(port.Name) == 0 {
+								if port.Port == 443 {
+									protocol = "https"
+								}
+							} else {
+								protocol = port.Name
```

Comment:
[ServicePort names must be unique](http://kubernetes.io/docs/api-reference/v1/definitions/#_v1_serviceport), so this will fail if there multiple ports that use the same scheme.

```
apiVersion: v1
kind: Service
metadata:
  labels:
    app: wip
    branch: andy
    environment: stage
  name: wip-stage-test
spec:
  ports:
    - name: http
      port: 80
    - name: http-alt
      port: 8080
    - name: https
      port: 443
    - name: https-alt
      port: 6443
  selector:
    app: wip
    branch: andy
    environment: stage
```